.github/workflows/template-build.yml

```diff
@@ -79,6 +79,36 @@ jobs:
 
       - run: yarn install
 
+      - name: Azure CLI - Login
+        uses: azure/login@v1
+        with:
+          client-id: ${{ secrets.AZURE_CLIENT_ID }}
+          tenant-id: ${{ secrets.AZURE_TENANT_ID }}
+          subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+
+      - name: KeyVault - Secrets
+        shell: pwsh
+        id: KeyVaultSecrets
+        run: |
+          # KV - Role Assignment to SP
+
+          az deployment group create `
+            --name ${{ env.KEY_VAULT }}-${{ steps.date.outputs.date }} `
+            --resource-group ${{ env.AZURE_RESOURCE_GROUP }} `
+            --template-file ./infra/keyVaultRoleAssignment.bicep `
+            --parameters `
+              "keyVaultName=${{ env.KEY_VAULT }}" `
+              "principalId=${{ secrets.AZURE_SERVICE_PRINCIPAL_OBJECT_ID }}" `
+              "principalType=ServicePrincipal" `
+              "roleName=Key Vault Secrets User" `
+            -o none
+          Write-Host '✅ SP - kv role assigned'
```

Comment:
this should ideally be done as part of the environment deployment

Reply:
I've added it in the bicep while env deployment.